src/shape.rs

```diff
@@ -1180,12 +1188,14 @@ impl ShapeLine {
                     let mut fitting_start = (0, 0);
```

Comment:
Confirming:

For the Warp case, I guess technically incongruent is possible too? The plan is to follow up with that? I remember we chatted about this a bit - just curious whether we need it for the Warp use case (or if congruent is enough for us)? I guess technically Warp users could mix-and-match LTR and RTL text in the same line (though I imagine this would be kinda hard to do haha 😅, curious to try this in Warp)

Aligned on definitely needing it for the upstream contribution but yep, just wanted to get a better understanding!

Reply:
I'm not sure whether we need it immediately, but I'm also not super familiar with how frequently users in different locales mix and match LTR/RTL text. If I had to guess, it's very common, so I will try to add support for it ASAP so that it's ready for the relevant features.

Reply:
Got it - thanks for elaborating!